content/articles/getting-started-with-unit-testing-with-spring-boot/index.md

```diff
@@ -0,0 +1,131 @@
+#### Loading functionality in slices
+It is crucial to restrict the application context to only the Spring components included in the test scenario when testing for an extensive application. This is achieved by including them in the annotation declaration.
+
+This particular feature, if used wisely, can help us build narrow tests without such a hefty penalty in terms of performance, particularly for small-sized apps.
+#### Use the `@DataJpaTest` Annotation
+To increase the performance of components. We use the `@DataJpaTest` annotation since it will not load beans annotated with`@Service`, `@Controller`, and the entire application context.
+#### Performing tests that are database-related
+It's good to simulate database-interacting beans and disable Spring Boot test DB initialization for the Spring profile where the tests are executed. When testing controllers, you should keep this in mind.
+#### Make your test simple
+Whenever an expert developer teaches unit testing to beginners, he should always ensure that the tests are correct and straightforward. To achieve this, he should keep the test with low cyclomatic complexity. Cyclomatic complexity is a coding statistic that shows how many different execution pathways a procedure can take. Developers are less likely to introduce problems when working on code with a lesser complexity since it is easier to comprehend and maintain.
+### Reasons for testing...
+#### Repositories
+The repository is the place where all the data is stored. It is tested to ensure that specs or relationships have been correctly implemented.
+#### Services
+This is the layer where all the business logic is implemented. It is tested to make sure that the business logic is correct.
+### How to test controllers
+Let us now look at how to test controllers in Spring Boot. To accomplish this, you will need to import some dependencies using the `Spring Initializer` and `IntelliJ IDEA.` `IntelliJ IDEA` lets us create a Spring Boot application using the Spring Initializr service. Open `file>new>project` and select `Spring Initializr` as shown below:
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/spring-Initializer.png)
+
+> Note: You can rename the Package name.
+
+Click next to proceed. Type *web* to search the required dependencies in the search bar. Select *Spring Web*, *Spring Reactive Web*, *Spring Web Services*, and click *finish* to download the initializr template:
+
+![Spring Initializer](/engineering-education/getting-started-with-unit-testing-with-spring-boot/spring-Initializer-Dependency.png)
+
+We've prepared the environment successfully. Now we need to test the Controller.
+In `/src/main/java` create a new package named `Controller`. Proceed to create a Java class named `HelloContoller.java` in  `/src/main/java/Controller`. Within that file, add the code below:
```

Comment:
package names should be lower case, i.e. the package should be named *controller* rather than *Controller*